.osc-patches/2022-04-05--config-add-helper-to-load-standard-configuration-fil.patch

```diff
@@ -0,0 +1,286 @@
+From 9c767013752489bf5c351a749b18a07a3f152c3f Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Jutteau?= <jerome.jutteau@outscale.com>
+Date: Tue, 5 Apr 2022 14:25:42 +0200
+Subject: [PATCH] config: add helper to load standard configuration file
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Signed-off-by: Jérôme Jutteau <jerome.jutteau@outscale.com>
+---
+ v2/config_file.go      | 126 ++++++++++++++++++++++++++++++++++++++++
+ v2/config_file_test.go | 129 +++++++++++++++++++++++++++++++++++++++++
+ 2 files changed, 255 insertions(+)
+ create mode 100644 v2/config_file.go
+ create mode 100644 v2/config_file_test.go
+
+diff --git a/v2/config_file.go b/v2/config_file.go
+new file mode 100644
+index 00000000..a0ca79f0
+--- /dev/null
++++ b/v2/config_file.go
+@@ -0,0 +1,126 @@
++package osc
++
++import (
++	"context"
++	"encoding/json"
++	"errors"
++	"fmt"
++	"io/ioutil"
++	"os"
++	"path"
++)
++
++type ConfigFile struct {
++	profiles map[string]Profile
++}
++
++type Profile struct {
++	AccessKey       string   `json:"access_key"`
++	SecretKey       string   `json:"secret_key"`
++	X509ClientCert  string   `json:"x509_client_cert"`
++	X509_client_key string   `json:"x509_client_key"`
++	Protocol        string   `json:"protocol"`
++	Method          string   `json:"method"`
++	Region          string   `json:"region"`
++	Endpoints       Endpoint `json:"endpoints"`
++}
++
++type Endpoint struct {
++	API string `json:"api"`
++	FCU string `json:"fcu"`
++	LBU string `json:"lbu"`
++	EIM string `json:"eim"`
++	ICU string `json:"icu"`
++	OOS string `json:"oos"`
++}
++
++func NewConfigFile() *ConfigFile {
++	return &ConfigFile{
++		profiles: make(map[string]Profile),
++	}
++}
++
++func defaultConfigPath() (string, error) {
++	home, err := os.UserHomeDir()
++	if err != nil {
++		return "", err
++	}
++	return path.Join(home, ".osc", "config.json"), nil
++}
++
++func LoadDefaultConfigFile() (*ConfigFile, error) {
++	configPath, err := defaultConfigPath()
++	if err != nil {
++		return nil, err
++	}
++	return LoadConfigFile(&configPath)
++}
++
++func LoadConfigFile(path *string) (*ConfigFile, error) {
```

Comment:
Is there a reason why we have a pointer instead of the value itself ? (other than to skip the copy of the data)

Reply:
Nope, no other reason than just data copy

Reply:
ok 👍